scripts/gen-readme.go

```diff
@@ -0,0 +1,252 @@
+		}
+
+		propertiesBytes, err := os.ReadFile(workflow.PropertiesPath)
+		if err != nil {
+			if os.IsNotExist(err) {
+				hasInvalidConfigs = true
+				fmt.Println(fmt.Sprintf("properties file does not exist for workflow %s: path - %s", workflowID, workflow.PropertiesPath))
+			} else {
+				return fmt.Errorf("failed to read properties file %s: %w", workflow.PropertiesPath, err)
+			}
+		}
+
+		if hasInvalidConfigs {
+			continue
+		}
+
+		var propertiesConfigs PropertiesConfig
+		if err := json.Unmarshal(propertiesBytes, &propertiesConfigs); err != nil {
+			return fmt.Errorf("failed to unmarshal properties file %s: %w", workflow.PropertiesPath, err)
+		}
+
+		parts := strings.Split(workflow.WorkflowPath, "/")
+
+		if len(parts) < 2 {
+			return errors.New("invalid workflow path %s, path should have at least 3 folders")
+		}
+
+		actionName := parts[1]
+		actionPath := path.Join(parts[:2]...)
+		actionReadMePath := path.Join(actionPath, "README.md")
+
+		if _, err := os.Stat(actionReadMePath); err != nil {
+			if os.IsNotExist(err) {
+				actionReadMeContents := fmt.Sprintf("# %s examples", actionName)
+				if err := os.WriteFile(actionReadMePath, []byte(actionReadMeContents), 0644); err != nil {
+					return fmt.Errorf("failed writing content to action README file %s: %w", actionReadMePath, err)
+				}
+			} else {
+				return fmt.Errorf("failed to validate %s README file exists: %w", actionReadMePath, err)
+			}
+		}
+
+		workflowSubPath := path.Join(parts[2:]...)
+		workflowRelativeName := strings.TrimSuffix(workflowSubPath, filepath.Ext(workflowSubPath))
+
+		actionData, hasKey := readmeActions[actionName]
+		if !hasKey {
+			emptyWorkflows := make([]ReadmeWorkflow, 0)
+			actionData = ReadmeAction{
+				Name:       actionName,
+				Path:       actionPath,
+				ReadMePath: actionReadMePath,
+				Workflows:  emptyWorkflows,
+			}
+		}
+
+		actionData.Workflows = append(actionData.Workflows, ReadmeWorkflow{
+			Name:           propertiesConfigs.Name,
+			RelativeName:   workflowRelativeName,
+			Description:    propertiesConfigs.Description,
+			Starter:        workflow.Starter,
+			WorkflowPath:   workflow.WorkflowPath,
+			PropertiesPath: workflow.PropertiesPath,
+		})
+
+		readmeActions[actionName] = actionData
+	}
+
+	sortedActions := getSortedActionNames(readmeActions)
+
+	readmeTemplateConfigs := ReadmeTemplateConfig{
+		Title:   "Google GitHub Actions - Example Workflows",
+		Actions: sortedActions,
+	}
+
+	if hasInvalidConfigs {
+		return fmt.Errorf("failed to process invalid configs")
+	}
+
+	if err := renderReadmeTemplate(readmeTemplateConfigs); err != nil {
+		return fmt.Errorf("failed to render readme template: %w", err)
+	}
+
+	return nil
+}
+
+func renderReadmeTemplate(templateConfig ReadmeTemplateConfig) error {
+	readmeTemplate, err := template.ParseFiles(readmeTmplatePath)
+	if err != nil {
+		return fmt.Errorf("failed to create properties template: %w", err)
```

Comment:
```suggestion
		return fmt.Errorf("failed to parse readme template: %w", err)
```

Reply:
IIUC errors below also refer to properties when it should be readme.

Reply:
ack